jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -407,11 +412,11 @@ def _ndarray_to_glue_data(arr, data_label):
 # ---- Functions that handle WCS-only data -----
 
 def _wcsonly2data(ndd, data_label):
-    """Return Data given NDData containly WCS-only data."""
+    """Return Data given NDData containing WCS-only data."""
     arr = ndd.data
     data = Data(label=data_label)
     data.meta.update(standardize_metadata(ndd.meta))
     data.coords = ndd.wcs
     component = Component.autotyped(arr, units="")
     data.add_component(component=component, label="DATA")
-    return data
+    yield (data, data_label)
```

Comment:
I didn't think it would be inside the parser, but now that it is, yes, it has to be a generator.

Also maybe rename to `_wcsonly_to_glue_data` to be consistent with other generator functions.